validation/validator.go

```diff
@@ -67,3 +85,178 @@ func (v *Validator) Errors() httpvalidate.Errors {
+	for tag, field := range tagToFieldMap {
+		if value, ok := values[tag]; ok && len(value) > 0 {
+			if err := v.setFieldValue(field, value[0]); err != nil {
+				return err
+			}
+		}
+		if formData != nil {
+			if value, ok := formData.Get(tag); ok {
+				if err := v.setFieldValue(field, value); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (v *Validator) bindFromMap(dataMap map[string]any, tagToFieldMap map[string]reflect.Value) error {
+	for tag, field := range tagToFieldMap {
+		if value, ok := dataMap[tag]; ok {
+			if err := v.setFieldValue(field, value); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (v *Validator) bindFromStruct(dataSrc reflect.Value, tagToFieldMap map[string]reflect.Value) error {
+	for tag, field := range tagToFieldMap {
+		if value := dataSrc.FieldByName(tag); value.IsValid() && value.CanInterface() {
+			if err := v.setFieldValue(field, value.Interface()); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (v *Validator) castValueToType(field reflect.Value, value any) (reflect.Value, error) {
+	var castedValue any
+	var err error
+
+	switch field.Kind() {
+	case reflect.String:
+		castedValue = cast.ToString(value)
+	case reflect.Int:
+		castedValue, err = cast.ToIntE(value)
+	case reflect.Int8:
+		castedValue, err = cast.ToInt8E(value)
+	case reflect.Int16:
+		castedValue, err = cast.ToInt16E(value)
+	case reflect.Int32:
+		castedValue, err = cast.ToInt32E(value)
+	case reflect.Int64:
+		castedValue, err = cast.ToInt64E(value)
+	case reflect.Uint:
+		castedValue, err = cast.ToUintE(value)
+	case reflect.Uint8:
+		castedValue, err = cast.ToUint8E(value)
+	case reflect.Uint16:
+		castedValue, err = cast.ToUint16E(value)
+	case reflect.Uint32:
+		castedValue, err = cast.ToUint32E(value)
+	case reflect.Uint64:
+		castedValue, err = cast.ToUint64E(value)
+	case reflect.Bool:
+		castedValue, err = cast.ToBoolE(value)
+	case reflect.Float32:
+		castedValue, err = cast.ToFloat32E(value)
+	case reflect.Float64:
+		castedValue, err = cast.ToFloat64E(value)
+	case reflect.Slice:
+		castedValue, err = cast.ToSliceE(value)
+	case reflect.Map:
+		castedValue, err = cast.ToStringMapE(value)
+	case reflect.Array:
+		castedValue, err = cast.ToSliceE(value)
+	case reflect.Struct:
+		structType := field.Type()
+		newStruct := reflect.New(structType).Elem()
+		err := v.populateStruct(newStruct, value)
+		if err != nil {
+			return reflect.Value{}, fmt.Errorf("%s: %w", errCastValueField, err)
+		}
+		castedValue = newStruct.Interface()
+	default:
+		castedValue = value
+	}
+
+	if err != nil {
+		return reflect.Value{}, fmt.Errorf("%s: %w", errCastValueField, err)
+	}
+
+	return reflect.ValueOf(castedValue), nil
+}
+
+func (v *Validator) populateStruct(structVal reflect.Value, valueMap any) error {
+	valueMapCasted, ok := valueMap.(map[string]any)
+	if !ok {
+		return fmt.Errorf("%s: %s", errCastValueMap, reflect.TypeOf(valueMap).String())
+	}
+
```

Comment:
This to support structure nesting, has passed the test in my project and CI test case.